bundler/lib/bundler/templates/newgem/circleci/config.yml.tt

```diff
@@ -3,11 +3,21 @@ jobs:
   build:
     docker:
       - image: ruby:<%= RUBY_VERSION %>
+<%- if config[:ext] == 'rust' -%>
+    environment:
+      RB_SYS_FORCE_INSTALL_RUST_TOOLCHAIN: 'true'
+<%- end -%>
     steps:
       - checkout
+<%- if config[:ext] == 'rust' -%>
+      - run:
+          name: Instal Rust/Cargo dependencies
+          command: apt-get update && apt-get install -y clang
+<%- end -%>
       - run:
           name: Run the default task
           command: |
+            gem update --system '<%= ::Gem.rubygems_version %>'
```

Comment:
The concern I was raising in your other PR is that (unless we enforce strictly the RubyGems >= 3.3.11 requirement and refuse to generate this file otherwise), this may not be modern enough to include the rust extensions feature.

Because of this, maybe we should go with a fixed high enough version like 3.3.26?

Reply:
You'll not be able to do anything useful locally as well wen generated by older RubyGems. Do we really need to handle those edge cases? Alt. I can check if local version is enough and prevent generator to provide Rust skeleton, since it will not be able to compile it.

Reply:
It would be possible to make it clear you need to install latest RubyGems to get best support.

Reply:
> Alt. I can check if local version is enough and prevent generator to provide Rust skeleton, since it will not be able to compile it.

Yes, this is exactly what I've been suggesting.

Reply:
Ahh, finally I do follow. I'll update.

Reply:
Sorry if I failed to explain my idea properly 🙏

Reply:
this should be handled at https://github.com/rubygems/rubygems/commit/68537383487afabc5793f35716a4e515ac7d6a67 now

Reply:
Awesome @simi, thanks!